samples/MemoryCacheFileWatchSample/Middleware/GreetingMiddleware.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using MemoryCacheFileWatchSample.Abstractions;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.AspNetCore.Hosting;
+using System.IO;
+using System;
+
+namespace MemoryCacheFileWatchSample
+{
+    public class GreetingMiddleware
+    {
+        private readonly IGreetingService _greetingService;
+        private readonly ILogger<GreetingMiddleware> _logger;
+        private readonly IMemoryCache _memoryCache;
+        private readonly RequestDelegate _next;
+        private readonly IHostingEnvironment _env;
+
+        public GreetingMiddleware(RequestDelegate next, IMemoryCache memoryCache, ILogger<GreetingMiddleware> logger, IGreetingService greetingService, IHostingEnvironment env)
+        {
+            _next = next;
+            _memoryCache = memoryCache;
+            _logger = logger;
+            _greetingService = greetingService;
+            _env = env;
+        }
+
+        public Task Invoke(HttpContext httpContext)
```

Comment:
Delete everything in this sample except this method body and move it to Program.Main